go/runtime/client/client.go

```diff
@@ -179,6 +191,7 @@ func (c *runtimeClient) SubmitTx(ctx context.Context, request *api.SubmitTxReque
 			continue
 
 		case resp, ok = <-respCh:
+			req.ExpectedEpochNumber = resp.epochNumber
```

Comment:
This seems problematic, you are creating a shared reference `req`, mutating it here and then passing the same reference to potentially different goroutines below (like 217, `go c.doSubmitTxToLeader`). This can cause problems, so instead only create a new `req` each time just before calling `go c.doSubmitTxToLoader`.